Event.js

```diff
@@ -0,0 +1,139 @@
+        if (!validParameters) {
+            return;
+        }
+
+        L.i(`recordEvent, Sending event: [eventName: ${eventName}, eventCount: ${eventCount}, eventSum: ${eventSum}, segments: ${segments}]`);
+
+        const args = {};
+        args.n = eventName;
+
+        if (eventCount) {
+            args.c = eventCount;
+        } else {
+            args.c = 1;
+        }
+
+        args.s = eventSum;
+
+        args.g = [];
+        for (const event in segments) {
+            args.g.push(event);
+            args.g.push(segments[event]);
+        }
+        this.#state.CountlyReactNative.recordEvent(args);
+    }
+
+    /**
+     *
+     * Start Event
+     * NB: If endEvent is not called (with the same event name),
+     * no event will be recorded.
+     *
+     * @param {string} eventName name of event
+     * @return {void} void
+     */
+    startEvent(eventName) {
+        if (!this.#state.isInitialized) {
+            L.d("startEvent, 'init' must be called before 'startEvent'");
+            return;
+        }
+        const isInvalid = Validate.String(eventName, "eventName", "startEvent");
+        if (isInvalid) {
+            return;
+        }
+        L.i(`startEvent, Starting event: [${eventName}]`);
+        this.#state.CountlyReactNative.startEvent([eventName.toString()]);
+    }
+
+    /**
+     *
+     * Cancels an Event
+     *
+     * @param {string} eventName name of event
+     * @return {void} void
+     */
+    cancelEvent(eventName) {
+        if (!this.#state.isInitialized) {
+            L.d("cancelEvent, 'init' must be called before 'cancelEvent'");
+            return msg;
+        }
+        const isInvalid = Validate.String(eventName, "eventName", "cancelEvent");
+        if (isInvalid) {
+            return;
+        }
+        L.i(`cancelEvent, Canceling event: [${eventName}]`);
+        this.#state.CountlyReactNative.cancelEvent([eventName.toString()]);
+    }
+
+    /**
+     *
+     * End Event
+     * NB: Should be called after startEvent.
+     *
+     * @param {string} eventName - Name of the event.
+     * @param {Segmentation} segments - segementation data for the event.
+     * @param {number} eventCount - event count.
+     * @param {number} eventSum - event sum.
+     * @return {void} void
+     */
+    endEvent(eventName, segments, eventCount, eventSum) {
+        if (!this.#state.isInitialized) {
+            L.d("endEvent, 'init' must be called before 'endEvent'");
+            return;
+        }
+        const validParameters = Validate.areEventParametersValid('endEvent', eventName, segments, eventCount, eventSum);
+        if (!validParameters) {
+            return;
+        }
+        L.i(`endEvent, Sending event: [eventName: ${eventName}, segments: ${segments}, eventCount: ${eventCount}, eventSum: ${eventSum}]`);
+
+        const args = {};
+        args.n = eventName;
+
+        if (eventCount) {
```

Comment:
and it should be positive

Reply:
`Validate.areEventParametersValid` now checks if eventCount is positive